### ISSUE TITLE
Replace overlapping events

### ISSUE THREAD

**Issue by PrinceOfAbyss** [Paul-DS/bootstrap-year-calendar#44]
I'll use this great calendar slightly differently from the majority of people. I'll be using it to define prices, like daily rates for some service. Ie, the operator of the system may choose a period, like 1/1/2016 - 31/3/2016, and assign a price tag for the period. Then, 1/4/2016 - 21/8/2016 another price. So, my intention is to achieve any new period that has overlapping dates with a pre-existing period to replace the existing prices. So, for the example above, if he then defines a period, like 15/1/2016 - 10/2/2016 and assign another price, I want that price replace the other two ones that it overlaps. Is there an easy way to achieve that? Ie, maybe the getEventsOnRange could do some sick magic and do most of the dirty part for me? Or, does it need to get done the hard way, meaning by that that I have to perform all the if's within the saveEvent method myself? If it's the latter, is there a proven algorithm to use? Cause I'm afraid that I might miss a rare case that'll start producing bugs in the longterm?

Thanks in advance for your reply.
Kind regards,
and thanks a million for this great app.

**Comment by noncototient** [comment]
If you are using a database, you can pass the data to the server and do the logic there and if price already exists for that date, simply replace it. 

And yes, there is no easy way to do it. You have to do everything the hard way, defining your own functions and logic.

**Comment by PrinceOfAbyss** [comment]
Yup, of course I'm storing the data in a database, and that was my original intention, to collect the data server-side and brake each period there, storing each day separately. So, I won't even bother check for overlapping periods. I'll be sorting the periods by ID, and store the price of each period for the appropriate day. Then, if another period along the line has some overlapping days, (having sorted them by ID) will make sure that the prices are the newer ones, so it'll be good.

While using the calendar, I won't have a problem either, as I'm using the background-color property instead of the underlining for periods, so the latest color will be visible at any moment.

Anyway, thanks for confirming my idea, and thanks for the super fast reply.

PS: I have two more issues, but I'll open separate topics for them.

**Comment by noncototient** [comment]
No problem, man! I like your idea and it seems you have all figured out so should be able to do it fairly easily ;)

**Comment by PrinceOfAbyss** [comment]
I put quite some thought into it! :)
